### PR TITLE
Remove gesture from page that interferes with accessibility

### LINKED ISSUE #11543: UISwitch doesn't toggle the first time if you have VoiceOver enabled
### Description

The code we use to try and auto close the keyboard when clicking off of an entry field currently causes the `UISwitch` to not toggle the first time you interact with it.

If you take the following code
```XAML
 <VerticalStackLayout>
        <Switch WidthRequest="50"></Switch>
    </VerticalStackLayout>
```

- Turn on VO
- Navigate to switch
- double tap
- nothing happens
- double tap again and now it works fine.

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

We were just adding a `TapGestureRecognizer` to every single instance of Page which would cause the gesture to always be "active" regardless of the state of the app.  This PR changes it so the gesture will only be added to the `UIWindow` when an entry field becomes a first responder. Once it loses first responder the gesture is removed from the window. This way the gesture is only active when it's relevant 

### Issues Fixed

Fixed #11543

### CODE REVIEW

File: src/Core/src/Handlers/Page/PageHandler.iOS.cs [hunk 1]
```diff
@@ -24,24 +24,6 @@ protected override ContentView CreatePlatformView()
 			throw new InvalidOperationException($"PageViewController.View must be a {nameof(ContentView)}");
 		}
 
-		protected override void ConnectHandler(ContentView nativeView)
-		{
-			var uiTapGestureRecognizer = new UITapGestureRecognizer(a => nativeView?.EndEditing(true));
```
Comment (jsuarezruiz) [inline src/Core/src/Handlers/Page/PageHandler.iOS.cs:29]: Should we include with this changes an alternative way to manage the keyboard etc?

File: src/Core/src/Platform/iOS/ResignFirstResponderTouchGestureRecognizer.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		IEnumerable<UIView> ViewAndSuperviewsOfView(UIView view)
+		{
+			while (view != null)
+			{
+				yield return view;
+				view = view.Superview;
+			}
+		}
+
+		internal static void Update(UITextView uIControl, UIWindow? window)
```
Comment (rachelkang) [inline src/Core/src/Platform/iOS/ResignFirstResponderTouchGestureRecognizer.cs:69]: I would suggest something like`textView` or `platformTextView` to distinguish from the following function


```suggestion
		internal static void Update(UITextView textView, UIWindow? window)
```

File: src/Core/src/Platform/iOS/ResignFirstResponderTouchGestureRecognizer.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+				uIControl.Ended += OnEditingDidEnd;
+			}
+			else
+			{
+				uIControl.Started -= OnEditingDidBegin;
+				uIControl.Ended -= OnEditingDidEnd;
+			}
+		}
+
+		internal static void Update(UIControl uIControl, UIWindow? window)
```
Comment (rachelkang) [inline src/Core/src/Platform/iOS/ResignFirstResponderTouchGestureRecognizer.cs:83]: trivial, but I think the `uIControl` casing can be a bit confusing

```suggestion
		internal static void Update(UIControl platformControl, UIWindow? window)
```

File: src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs [hunk 4]
```diff
@@ -59,6 +59,31 @@ public static async Task WaitForFocused(this AView view, int timeout = 1000)
...
+			if (view.IsFocused)
+			{
+				TaskCompletionSource focusSource = new TaskCompletionSource();
+				view.FocusChange += OnFocused;
+				await focusSource.Task.WaitAsync(TimeSpan.FromMilliseconds(timeout));
+
+				// Even thuogh the event fires unfocus hasn't fully been achieved
+				await Task.Delay(10);
+
+				void OnFocused(object? sender, AView.FocusChangeEventArgs e)
```
Comment (rachelkang) [inline src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs:82]:
```suggestion
				void OnUnFocused(object? sender, AView.FocusChangeEventArgs e)
```

File: src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs [hunk 5]
```diff
@@ -59,6 +59,31 @@ public static async Task WaitForFocused(this AView view, int timeout = 1000)
...
+				}
+			}
+		}
+
+		public static async Task WaitForUnFocused(this AView view, int timeout = 1000)
+		{
+			if (view.IsFocused)
+			{
+				TaskCompletionSource focusSource = new TaskCompletionSource();
+				view.FocusChange += OnFocused;
```
Comment (rachelkang) [inline src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs:76]:
```suggestion
				view.FocusChange += OnUnFocused;
```

File: src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs [hunk 6]
```diff
@@ -59,6 +59,31 @@ public static async Task WaitForFocused(this AView view, int timeout = 1000)
...
+				await focusSource.Task.WaitAsync(TimeSpan.FromMilliseconds(timeout));
+
+				// Even thuogh the event fires unfocus hasn't fully been achieved
```
Comment (rachelkang) [inline src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs:79]:
```suggestion
				// Even though the event fires unfocus hasn't fully been achieved
```

**Review by rachelkang — CHANGES_REQUESTED** [review]
Looks great to me! Switch issue is definitely no longer.

Just left a couple of nitpick comments

**Review by rachelkang — APPROVED** [review]
LGTM!

**Pull request merged**